hydro_deploy/core/src/util.rs

```diff
@@ -1,18 +1,16 @@
-use std::io;
 use std::sync::{Arc, Mutex};
 use std::time::Duration;
 
-use anyhow::Result;
 use futures::{Future, Stream, StreamExt};
 use tokio::sync::oneshot;
 
 use crate::ssh::PrefixFilteredChannel;
 
-pub async fn async_retry<T, F: Future<Output = Result<T>>>(
+pub async fn async_retry<T, E, F: Future<Output = Result<T, E>>>(
```

Comment:
where is `E` used?

Reply:
Previously this only accepted `Result<T, anyhow::Error>`, this change generalizes it to work with any error type hence the `E`. (Specifically, this file had `use anyhow::Result;` before above, which is an alias for `Result<T, anyhow::Error>`)